src/components/commons/QrCode.tsx

```diff
@@ -8,19 +8,23 @@ import QRCode from "react-qr-code";
 import { truncateTextFromMiddle } from "shared/textHelper";
 
 interface QrCodeProps {
+  href: string;
   address: string;
   onCloseClick: Dispatch<SetStateAction<boolean>>;
 }
 
-export default function WalletAddressQRCode({
-  address,
-  onCloseClick,
-}: QrCodeProps) {
+export default function QrCode({ href, address, onCloseClick }: QrCodeProps) {
   const [isWalletAddressCopied, setIsWalletAddressCopied] = useState(false);
+  const fullPath =
```

Comment:
i dont think we need to construct our own path, can you try to explore if `next/router` is applicable for this use case? 
ref: https://nextjs.org/docs/api-reference/next/router#router-object

Reply:
checked the docs `next/router` and checked out the `pathname` and `asPath `
these would return the paths eg `/contract/1x1c19380x0c19380x0c1938` or `/contract/[cid]` but wouldnt give the 
full path with eg `https://deploy-preview-79--defimetascan.netlify.app/` 
` window.location.href` would return the fullpath name which includes the `https://deploy-preview-79--defimetascan.netlify.app/` needed for the generation of the QrCode image